app/app/Http/Controllers/PasswordRecoveryController.php

```diff
@@ -83,6 +85,7 @@ public function resetPassword(Request $request)
                 return Redirect::back()->withErrors($validator)->withInput();
             } else {
                 $user->password_hash = User::generateSaltedHash($request->input('new_password'));
+                $user->email_verification_time = date('Y-m-d H:i:s');
```

Comment:
Wrap that line in a check to see that email_verification_time is null.  We don't want to assign a new value unless it is needed.  We want to remember when the account was initially activated.  Replacing it unconditionally like this will erase the original email verification time.